doc/boot_protocol.rst

```diff
@@ -0,0 +1,274 @@
+
+The Kernel Log tag provides a means to control/configure logging done by SKL
+itself. Specifically to enable/disable and set priority of the logging to be
+done by SKL.
+::
+
+    struct skl_tag_crtm_hash {
+        struct skl_tag_hdr hdr;
+        u8 enable;
+        u8 priority;
+    } __packed;
+
+MSB Class
+---------
+
+The Measured Secure Boot (MSB) is an upcoming optional operational mode of the
+SKL. It allows for the SKL to function as the Root of Trust behind a signing
+key which can be used to start a SecureBoot RTV trust chain. The MSB Class is
+for all tags relating to configuring MSB.
+
+MSB Key Tag
+^^^^^^^^^^^
+
+The MSK Key tag is to allow embedding an MSB key directly into SKL.
+::
+
+    struct skl_tag_msb_key {
+        struct skl_tag_hdr hdr;
+        u16 algo_id;
+        u8 key[];
+    } __packed;
+
+MSB Key DB Tag
+^^^^^^^^^^^^^^
+
+The MSK Key DB tag points to a key db list external to the SKL along with a
+hash of the db list.
+::
+
+    struct skl_tag_msb_keydb {
+        struct skl_tag_hdr hdr;
+        u32 keydb;
+        u32 size;
+        u16 algo_id;
+        u8 digest[];
+    } __packed;
+
+IOMMU Table
+===========
+
+The IOMMU Table will be constructed at the first 4k page-aligned address after
+the Boot Tags but no further than 61440 bytes into the SKL. 
+
+Log Buffer
+==========
+
+The Log Buffer region contains the TPM Event Log and optionally a kernel log
+for SLK status and debug messaging. Unless an Event Log Tag is provided to set
+the size of the Event Log, the Log Buffer region will be split evenly, 2kb each
+for the default allocation of 4kb.
+
+TPM Event Log
+-------------
+
+The TPM Event Log is required and will be located at the beginning of the Log
+Buffer region.
+
+CRTM Log Event
+^^^^^^^^^^^^^^
+
+The measurement made by the CPU as a result of the SKINIT instruction, the CRTM
+measurement, is recorded directly to the TPM and is not provided to the SLB,
+SLK in this case.  While not required, it is generally common practice to
+provide a TPM eventlog when attesting the values in the PCRs. To support these
+use cases it is necessary that the CRTM measurement be recorded in the log. As
+mentioned, since the CPU does not make the measurement it takes available, the
+SLK must receive or obtain the measurement value by another means. The SLK will
+support two methods, one will be for the DRTM Preamble to record the
+measurement of the SLK in a CRTM Hash boot tag and the second will be to read
+the DRTM PCR, PCR 17, at start up and record the value into the TPM Eventlog.
+The SLK will record in the event data of the log event a marker to signify the
+measurement value's source, this is important as it will change how the
+attester must replay the log to match a TPM quote.
+
+Kernel Log
+----------
+
+The kernel log is an optional log region for the SLK to store information and
+debugging messages. This log will be located after the Event Log and will be
+given the remainder of the Log Buffer region.
```

Comment:
Surely we want to have explicit base (or offset from the start of SKL) and length fields for the logs?

Reply:
Yah, it will need be otherwise it would require external entities to know/understand the memory layout.